src/Sql/Sql/Auditing/Cmdlet/SetAzSqlDatabaseAuditPolicy.cs

```diff
@@ -0,0 +1,187 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Sql.Auditing.Model;
+using Microsoft.Azure.Commands.Sql.Common;
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.Auditing.Cmdlet
+{
+    [Cmdlet(
+        VerbsCommon.Set,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + DefinitionsCommon.DatabaseAuditPolicyCmdletsSuffix,
+        DefaultParameterSetName = DefinitionsCommon.DatabaseParameterSetName,
+        SupportsShouldProcess = true),
+        OutputType(typeof(bool))]
+    public class SetAzSqlDatabaseAuditPolicy : SqlDatabaseAuditPolicyCmdlet
+    {
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = AuditingHelpMessages.AuditActionGroupsHelpMessage)]
+        public AuditActionGroups[] AuditActionGroup { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = AuditingHelpMessages.AuditActionHelpMessage)]
+        public string[] AuditAction { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = AuditingHelpMessages.PredicateExpressionHelpMessage)]
+        [ValidateNotNull]
+        public string PredicateExpression { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = AuditingHelpMessages.BlobStorageTargetState)]
+        [ValidateSet(SecurityConstants.Enabled, SecurityConstants.Disabled, IgnoreCase = false)]
```

Comment:
@bashahee in the cases where you're using `ValidateSet`, we advise to use [`PSArgumentCompleter`](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/design-guidelines/parameter-best-practices.md#generic-argument-completer) instead as this provides the same tab fuctionality but allows users to provide values outside the set in the case that the API was updated to support additional values but the cmdlets haven't been updated

Reply:
"For any parameter which you would like the user to tab through a list of suggested values (but you do not want to limit the users to only these values), the generic argument completer should be added."
In our case, we do want to limit the users to the values in the set. Moreover, this value is not being passed as is to the REST-API.